staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation.go

```diff
@@ -191,6 +193,24 @@ func validateKMSCacheSize(c *config.KMSConfiguration, fieldPath *field.Path) fie
 	return allErrs
 }
 
+func validateKMSCacheHealthyTTL(c *config.KMSConfiguration, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.CacheHealthyTTL.Duration < 0 {
+		allErrs = append(allErrs, field.Invalid(fieldPath, c.CacheHealthyTTL, fmt.Sprintf(negativeValueErrFmt, "cacheHealthyTTL")))
+	}
+
+	return allErrs
+}
+
+func validateKMSCacheUnHealthyTTL(c *config.KMSConfiguration, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.CacheUnHealthyTTL.Duration < 0 {
+		allErrs = append(allErrs, field.Invalid(fieldPath, c.CacheUnHealthyTTL, fmt.Sprintf(negativeValueErrFmt, "cacheUnHealthyTTL")))
+	}
+
+	return allErrs
+}
```

Comment:
can these be combined?

Reply:
The certainly could be, but I think readability would suffer a bit.